toxcore/network.c

```diff
@@ -207,8 +207,10 @@ static int inet_pton4(const char *addrString, struct in_addr *addrbuf)
     struct sockaddr_in saddr = {0};
 
     INT len = sizeof(saddr);
+    char addrStringMut[128] = {0};
+    memcpy(addrStringMut, addrString, min_u16(sizeof(addrStringMut) - 1, strlen(addrString)));
```

Comment:
Why is this intermediate buffer needed? `WSAStringToAddress()`'s 1st argument is marked as `[in]`, not `[in, out]`, so it shouldn't mutate it. Just casting the const away, like what it was doing with the (LPTSTR) cast, should be fine.

Reply:
That means we'll have to disable `-Wcast-qual` globally or at least for this file, which is a bit annoying (but possible). Should we do that?

Reply:
Done.

Reply:
LGTM?

Reply:
>That means we'll have to disable -Wcast-qual globally or at least for this file, which is a bit annoying

I see, the intermediate buffer sounds good then.